main.tf

```diff
@@ -34,19 +34,18 @@ resource "google_project_iam_member" "token_creator_binding" {
 }
 
 resource "google_pubsub_topic_iam_member" "push_topic_binding" {
-  for_each = var.create_topic ? { for i in var.push_subscriptions : i.name => i } : {}
-
-  project = var.project_id
-  topic   = lookup(each.value, "dead_letter_topic", "projects/${var.project_id}/topics/${var.topic}")
-  role    = "roles/pubsub.publisher"
-  member  = "serviceAccount:${local.pubsub_svc_account_email}"
+  for_each = { for i in var.push_subscriptions : i.name => i if lookup(i, "dead_letter_topic", null) != null }
```

Comment:
This PR seems to be doing several other things at the same time, I'm not sure what this change is intended for?

Reply:
Because now we can set the `create_topic = false`, while still able to create the subscription.   
This iam role is ONLY needed when we set the dead_letter_topic, so if the subscription does not have it, it should not be created.